docs/protocol.html

```diff
@@ -0,0 +1,163 @@
+<h3><a id="protocol" href="#protocol">Kafka Wire Protocol</a></h3>
+
+<p>This document covers the protocol implemented in Kafka 0.8 and beyond. It is meant to give a readable guide to the protocol that covers the available requests, their binary format, and the proper way to make use of them to implement a client. This document assumes you understand the basic design and terminology described <a href="https://kafka.apache.org/documentation.html#design">here</a></p>
```

Comment:
Since this page is now in the version control and generated, I'm not sure this makes sense... we'll have a separate generated doc with each release, documenting the protocol for current version.